dali/operators/decoder/nvjpeg/legacy_api/nvjpeg_decoder.h

```diff
@@ -301,22 +301,22 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
             GetOutputPitch(output_type_) * info.widths[0];
         }
 
-        thread_pool_.DoWorkWithID(std::bind(
-              [this, info, data, in_size, output_data, file_name](int idx, int tid) {
-                DecodeSingleSampleHost(idx,
-                                       batched_image_idx_[idx],
-                                       tid,
-                                       handle_,
-                                       states_[0],
-                                       info,
-                                       data, in_size,
-                                       output_data,
-                                       streams_[0],
-                                       file_name);
-              }, i, std::placeholders::_1));
+        thread_pool_.AddWork(
+            [this, info, data, in_size, output_data, file_name, i](int tid) {
+              DecodeSingleSampleHost(idx,
+                                      batched_image_idx_[i],
+                                      tid,
+                                      handle_,
+                                      states_[0],
+                                      info,
+                                      data, in_size,
+                                      output_data,
+                                      streams_[0],
+                                      file_name);
+            }, -i);  // -i for FIFO order
```

Comment:
Maybe add a comment that samples are sorted already and you wan to preserve that order.